number.go

```diff
@@ -400,7 +401,7 @@ func parseInt[T integer](s string) (T, error) {
 }
 
 func parseUint[T unsigned](s string) (T, error) {
-	v, err := strconv.ParseUint(trimZeroDecimal(s), 0, 0)
+	v, err := strconv.ParseUint(trimDecimal(s), 0, 0)
```

Comment:
I'm suspicious here

It works for the expected case. The one that caused to code this.

But what about the string "10...17" previously it was returning an error, now it will happily return 10

https://go.dev/play/p/bkiLc4jdBEH

vs

https://go.dev/play/p/VyqcbrcMEWk

The issue is worst if you consider:
- 10.foobar (invalid)
- 10.0i (a complex)
- 10.0E9 (unsupported syntax previously)

All of them are now returning 10

Reply:
Fair enough. Let me try to fix that.